src/internal/connector/exchange/data_collections.go

```diff
@@ -0,0 +1,61 @@
+package exchange
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/data"
+)
+
+// ParseMetadataCollections produces two maps:
+// 1- paths: folderID->filePath, used to look up previous folder pathing
+// in case of a name change or relocation.
+// 2- deltas: folderID->deltaToken, used to look up previous delta token
+// retrievals.
+func ParseMetadataCollections(
+	ctx context.Context,
+	colls []data.Collection,
+) (map[string]string, map[string]string, error) {
```

Comment:
I think this will work, though if we want stricter checking (or if we're worried about folder ID collisions) we may want to make these `map[path.CategoryType]map[string]string`

Reply:
Can't say I've seen any reason to worry over folder id collision (maybe you know of something?).  I'm not against strictness, but I think that in this case the additional complexity outweighs the value gained.